manifests/run_service.pp

```diff
@@ -24,14 +24,47 @@
     }
   }
 
+  if ($::consul::manage_service == true) and ($::consul::selected_install_method == 'docker') {
+    $server_mode = pick($::consul::config_hash[server], false)
+      
+      if $server_mode {
+        $env = [ '\'CONSUL_LOCAL_CONFIG={"skip_leave_on_interrupt": true}\'', '\'CONSUL_ALLOW_PRIVILEGED_PORTS=\'']
+        $command = "agent -server"
+      }
+      else {
+        $env = [ '\'CONSUL_LOCAL_CONFIG={"leave_on_terminate": true}\'' ]
+        $command = "agent"
+      }
+
+      # Docker Install
+      docker::run { 'consul' :
+        image   => "${consul::docker_image}:${consul::version}",
+        net     => 'host',
+        volumes => [ "${::consul::config_dir}:/consul/config" ],
+        env     => $env,
+        command => $command
+      }
+  }
+
   if $::consul::join_wan {
     exec { 'join consul wan':
       cwd       => $::consul::config_dir,
       path      => [$::consul::bin_dir,'/bin','/usr/bin'],
       command   => "consul join -wan ${consul::join_wan}",
+      onlyif    => "/usr/bin/test ! -e ${consul::config_dir}/docker_used",
```

Comment:
probably need to assemble $command variable then add a prefix if install_method == docker.  If you look close, in order to run consul in the container I need to call it slightly differently (prefix it with "docker exec -t consul <...>").  So ... yeah if we "veto auto" we can get away from that.